### PR TITLE
Use ruff for linting

### PR DESCRIPTION
**Pull request by lymanepp**
HA transitioned to ruff a while ago. It's much faster than pylint and also has black and isort capabilities

### CODE REVIEW

File: pynws/__init__.py [hunk 1]
```diff
@@ -4,5 +4,8 @@
 """
 
 from .const import version
-from .nws import *
```
Comment (lymanepp) [inline pynws/__init__.py:7]: ruff doesn't like this and the fix doesn't over-export implementation details.

**Comment by MatthewFlamm** [comment]
Should this be in pre-commit?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    "SLOT", # flake8-slots
+    "T100", # Trace found: {name} used
+    "T20", # flake8-print
+    "TID251", # Banned imports
+    "TRY", # tryceratops
+    "UP", # pyupgrade
+    "W", # pycodestyle
+]
+
+ignore = [
```
Comment (lymanepp) [inline pyproject.toml:77]: Goal should be to un-ignore these over time

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,95 @@
+[tool.ruff]
+required-version = ">=0.3.7"
+target-version = "py38"
+
+[tool.ruff.lint]
+select = [
```
Comment (lymanepp) [inline pyproject.toml:6]: This list was copied from HA

**Comment by lymanepp** [comment]
> Should this be in pre-commit?

Done

**Comment by MatthewFlamm** [comment]
I'm +1 for this, the  pre-commit is being run against all files while the test is being run only against `pynws`.

A bunch of failures in setup.py.  We should ignore these as my plan is to move to `setuptoole_scm` and configuration in `pyproject.toml`.

**Comment by lymanepp** [comment]
> A bunch of failures in setup.py. We should ignore these...

These have been fixed.

**Review by MatthewFlamm — APPROVED** [review]
LGTM

**Pull request merged**